cdsapi/api.py

```diff
@@ -62,6 +64,24 @@ def toJSON(obj):
     return obj
 
 
+def _open_target(target: str | bytes | os.PathLike | io.RawIOBase | io.BufferedIOBase) -> IO[bytes]:
+    """Creates a BinaryIO context manager given a certain destination file target."""
+    ERR_MSG = "Invalid value for target. Use None, path string or bytes, or a file-like object."
+
+    if isinstance(target, (str, bytes, os.PathLike)):
+        file = open(target, "wb")
+    elif isinstance(target, (io.RawIOBase, io.BufferedIOBase)):
+        try:
+            file = target
+            file.seek(0)
```

Comment:
make sure we start to write at the start of the file for accurate download size counting